compiler/src/iree/compiler/Codegen/Common/ConvertToDestinationPassingStylePass.cpp

```diff
@@ -315,70 +315,153 @@ static bool canUseInOperandAsOutOperand(
+                                                     OpOperand *outOperand) {
+  /// Currently handle only uses as `ins` of `linalg.generic`. Replace the
+  /// `outOperand` with the `inOperand`, and drop its use from the `ins` operand
+  /// list.
+  auto genericOp = cast<linalg::GenericOp>(inOperand->getOwner());
+  assert(genericOp == outOperand->getOwner() &&
+         "expected in operand and out operand to be the same op");
+  SmallVector<Value> newInputs;
+  SmallVector<Value> newOutputs;
+  SmallVector<Type> newResultTypes;
+  SmallVector<AffineMap> maps;
+  for (OpOperand *in : genericOp.getDpsInputOperands()) {
+    if (in != inOperand) {
+      newInputs.push_back(in->get());
+      maps.push_back(genericOp.getMatchingIndexingMap(in));
+    }
+  }
+  for (OpOperand *out : genericOp.getDpsInitOperands()) {
+    maps.push_back(genericOp.getMatchingIndexingMap(out));
+    if (outOperand == out) {
+      newOutputs.push_back(inOperand->get());
+      newResultTypes.push_back(inOperand->get().getType());
+    } else {
+      newOutputs.push_back(out->get());
+      newResultTypes.push_back(out->get().getType());
+    }
   }
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPoint(genericOp);
+
+  Location loc = genericOp.getLoc();
+  SmallVector<utils::IteratorType> iterTypes(genericOp.getNumLoops(),
+                                             utils::IteratorType::parallel);
+  auto newOp = rewriter.create<linalg::GenericOp>(
+      loc, newResultTypes, newInputs, newOutputs, maps, iterTypes,
+      /*bodyBuild=*/nullptr, linalg::getPrunedAttributeList(genericOp));
+  rewriter.inlineRegionBefore(genericOp.getRegion(), newOp.getRegion(),
+                              newOp.getRegion().begin());
+
+  // Repair the payload entry block.
+  Block &payload = newOp.getRegion().front();
+  payload.getArgument(inOperand->getOperandNumber())
+      .replaceAllUsesWith(payload.getArgument(outOperand->getOperandNumber()));
+  payload.eraseArgument(inOperand->getOperandNumber());
+
+  rewriter.replaceOp(genericOp, newOp.getResults());
+  return success();
+}
 
- private:
-  bool useWARForCooperativeMatrixCodegen;
-};
+/// For results of compute ops that are not part of a store set, tries to modify
+/// its users to get the result into a store set, avoiding the use of stack
+/// allocation. This is done by looking for users
+/// 1) The result of the user is in the store set.
+/// 2) The use of the result of the compute op can be updated such that
+///    the new use is tied to the result of the user.
+/// This makes the result of the compute op be in the store set, and
+/// bufferizable without using a new stack. See
+/// https://github.com/iree-org/iree/issues/8303.
+static LogicalResult adaptComputeConsumerToAvoidStackAllocation(
+    func::FuncOp funcOp, bool useWARForCooperativeMatrixCodegen) {
+  IRRewriter rewriter(funcOp.getContext());
+
+  constexpr int maxNumIterations = 6;
```

Comment:
style nit: `kMaxNumIterations` for constants.

IIRC, we are following google style guide for variable naming: https://google.github.io/styleguide/cppguide.html#Constant_Names